alerta/utils/config.py

```diff
@@ -1,6 +1,130 @@
 import os
-
 from flask import Flask
+from voluptuous import Schema, Url, Email
+import logging
+import ast
+
+
+LOG = logging.getLogger('alerta.config')
+
+
+class Validate:
+    def __init__(self):
+        self.string_validator = Schema(str)
+        self.integer_validator = Schema(int)
+        self.list_validator = Schema(list)
+        self.url_validator = Schema(Url())
+        self.email_validator = Schema(Email())
+
+    def get_variable(self, variable_name):
+        environment_string = os.environ[variable_name]
+
+        return environment_string
+
+    def validate_boolean(self, variable_name, boolean_string):
+        if boolean_string == 'True' or boolean_string == 'true' or boolean_string == '1':
+            variable_value = True
+            LOG.info(
+                f'Environment variable {variable_name} parsed with value {variable_value}')
+
+        elif boolean_string == 'False' or boolean_string == 'false' or boolean_string == '0':
+            variable_value = False
+            LOG.info(
+                f'Environment variable {variable_name} parsed with value {variable_value}')
+
+        else:
+            variable_value = False
+            LOG.error(
+                f'Environment variable {variable_name} parsed with value {variable_value}')
+
+        return variable_value
+
+    def validate_url(self, variable_name, url_string):
+        try:
+            if isinstance(url_string, list):
+                variable_value = []
+                for item in url_string:
+                    self.url_validator(item)
+                variable_value = url_string
+
+            else:
+                variable_value = self.url_validator(url_string)
+
+            LOG.info(
+                f'Environment variable {variable_name} parsed with value {variable_value}')
+            return variable_value
+
+        except Exception as e:
+            LOG.error(
+                f'Unable to parse environment variable {variable_name} {str(e)}')
+            return None
+
+    def validate_string(self, variable_name, string_string):
```

Comment:
I definitely think you should be using type annotations for all the methods as you are literally doing type checking of environment variables. For example, this should be ...
```
def validate_string(self, variable_name: str, string_string: str) -> str:
```

Reply:
Will update